### PR TITLE
Migrate tests to use Truth framework (first round)

### PR DESCRIPTION
**Pull request by chanseokoh**
First round of changes.

### CODE REVIEW

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 1]
```diff
@@ -162,16 +170,15 @@ public void testPluginConfigurationProcessor_defaults()
           InvalidCreationTimeException {
     BuildContext buildContext = getBuildContext(processCommonConfiguration());
 
-    Assert.assertEquals(
+    assertEquals(
         Arrays.asList("java", "-cp", "/app/resources:/app/classes:/app/libs/*", "java.lang.Object"),
         buildContext.getContainerConfiguration().getEntrypoint());
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:175]:
```suggestion
    assertThat(buildContext.getContainerConfiguration().getEntrypoint()).isEqualTo(ImmutableList.of("java", "-cp", "/app/resources:/app/classes:/app/libs/*", "java.lang.Object");

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 2]
```diff
@@ -224,8 +231,8 @@ public void testPluginConfigurationProcessor_extraDirectory()
                 layerEntry ->
                     layerEntry.getExtractionPath().equals(AbsoluteUnixPath.get("/target/dir/foo")))
             .findFirst();
-    Assert.assertTrue(fooEntry.isPresent());
-    Assert.assertEquals("123", fooEntry.get().getPermissions().toOctalString());
+    assertThat(fooEntry).isPresent();
+    assertEquals("123", fooEntry.get().getPermissions().toOctalString());
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:235]:
```suggestion
    assertThat(fooEntry.get().getPermissions().toOctalString()).isEqualTo("123");
```

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 3]
```diff
@@ -252,12 +259,12 @@ public void testEntrypoint()
...
           InvalidContainerizingModeException, InvalidFilesModificationTimeException,
           InvalidCreationTimeException {
-    Mockito.when(rawConfiguration.getEntrypoint())
+    when(rawConfiguration.getEntrypoint())
         .thenReturn(Optional.of(Arrays.asList("custom", "entrypoint")));
 
     BuildContext buildContext = getBuildContext(processCommonConfiguration());
 
-    Assert.assertEquals(
+    assertEquals(
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:267]: `assertThat().isEqualTo()`

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 4]
```diff
@@ -903,25 +905,25 @@ public void testGetJavaContainerBuilderWithBaseImage_incompatibleJava11JettyBase
...
     try {
       PluginConfigurationProcessor.getJavaContainerBuilderWithBaseImage(
           rawConfiguration, projectProperties, inferredAuthProvider);
-      Assert.fail();
+      fail();
     } catch (IncompatibleBaseImageJavaVersionException ex) {
-      Assert.assertEquals(11, ex.getBaseImageMajorJavaVersion());
-      Assert.assertEquals(12, ex.getProjectMajorJavaVersion());
+      assertEquals(11, ex.getBaseImageMajorJavaVersion());
+      assertEquals(12, ex.getProjectMajorJavaVersion());
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:926]: let's use `assertThrows` here?

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 5]
```diff
@@ -129,27 +138,26 @@ private TestPlatformConfiguration(@Nullable String architecture, @Nullable Strin
 
   @Before
   public void setUp() throws IOException, InvalidImageReferenceException, InferredAuthException {
-    Mockito.when(rawConfiguration.getFromAuth()).thenReturn(authProperty);
-    Mockito.when(rawConfiguration.getEntrypoint()).thenReturn(Optional.empty());
-    Mockito.when(rawConfiguration.getAppRoot()).thenReturn("/app");
+    when(rawConfiguration.getFromAuth()).thenReturn(authProperty);
+    when(rawConfiguration.getEntrypoint()).thenReturn(Optional.empty());
+    when(rawConfiguration.getAppRoot()).thenReturn("/app");
     Mockito.<List<?>>when(rawConfiguration.getPlatforms())
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:144]: hm what happens if we just call `when` in line 135?

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java [hunk 6]
```diff
@@ -162,16 +170,15 @@ public void testPluginConfigurationProcessor_defaults()
...
 
-    Mockito.verify(containerizer)
-        .setBaseImageLayersCache(Containerizer.DEFAULT_BASE_CACHE_DIRECTORY);
-    Mockito.verify(containerizer).setApplicationLayersCache(Paths.get("cache"));
+    verify(containerizer).setBaseImageLayersCache(Containerizer.DEFAULT_BASE_CACHE_DIRECTORY);
+    verify(containerizer).setApplicationLayersCache(Paths.get("cache"));
 
     ArgumentMatcher<LogEvent> isLogWarn = logEvent -> logEvent.getLevel() == LogEvent.Level.WARN;
-    Mockito.verify(logger, Mockito.never()).accept(Mockito.argThat(isLogWarn));
+    verify(logger, Mockito.never()).accept(Mockito.argThat(isLogWarn));
```
Comment (mpeddada1) [inline jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:181]: Mockito methods can usually be static imports

**Comment by chanseokoh** [comment]
@mpeddada1 these will be addressed in #3169.

File: jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java (hunk 5 above)
Comment (chanseokoh) [reply jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java:144]: Looks like type information is needed to resolve an incompatible generic type error.

**Review by mpeddada1 — APPROVED** [review]
Sounds good.

**Pull request merged**